cql3/statements/describe_statement.cc

```diff
@@ -396,13 +403,135 @@ class element_listing_describe_statement : public describe_statement {
     }
 };
 
+class generic_describe_statement final : public describe_statement {
+public:
+    using describe_statement::describe_statement;
+
+private:
+    std::optional<sstring> _keyspace;
+    sstring _name;
+    ::shared_ptr<selection::selection> _selection;
+    bool _with_internals;
+
+    sstring get_keyspace(const data_dictionary::database& db, const service::client_state& client_state) const {
+        if (!_keyspace) {
+            if (db.has_keyspace(_name)) {
+                return _name;
+            }
+
+            auto ks = client_state.get_raw_keyspace();
+            if (ks.empty()) {
+                throw exceptions::invalid_request_exception(format("'{}' not found in keyspaces", _name));
+            } else {
+                return ks;
+            }
+        } else {
+            if (db.has_keyspace(*_keyspace)) {
+                return *_keyspace;
+            } else {
+                throw exceptions::invalid_request_exception(format("'{}' not found in keyspaces", *_keyspace));
+            }
+        }
+    }
+
+    sstring get_type_name(const data_dictionary::database& db, const sstring& keyspace) const {
+        if (!_keyspace && db.has_keyspace(_name)) {
+            return raw::describe_statement::KEYSPACE;
+        }
+
+        auto table = db.try_find_table(keyspace, _name);
+        if (table) {
+            if (!table->schema()->is_view()) {
+                return raw::describe_statement::TABLE;
+            } else {
+                return raw::describe_statement::VIEW;
+            }
+        }
+
+        auto index = db.find_indexed_table(keyspace, _name);
+        if (index) {
+            return raw::describe_statement::INDEX;
+        }
+
+        throw exceptions::invalid_request_exception(format("'{}' not found in keyspace '{}'", _name, keyspace));
+    }
+
+protected:
+    virtual schema_ptr get_schema(const data_dictionary::database& db) const override {
+        return db.find_schema(KS, statements::describe_statement::ELEMENTS_TABLE);
+    }
+
+    virtual query::partition_slice get_slice(const data_dictionary::database& db, const service::client_state& client_state) const override {
+        auto schema = get_schema(db);
+        auto keyspace = get_keyspace(db, client_state);
+        auto type = get_type_name(db, keyspace);
+
+        partition_slice_builder slice_builder(*schema);
+        if (type != raw::describe_statement::KEYSPACE) {
+            slice_builder.with_range(query::clustering_range::make_singular(
+                clustering_key_prefix::from_exploded(*schema, {
+                    data_value(type).serialize_nonnull(),
+                    data_value(_name).serialize_nonnull()
+                })
+            ));
+        }
+        if(_with_internals) {
```

Comment:
`if` is not a function, so it wants a space after it.

Reply:
fixed